components/proxy-service/internal/proxy/unauthorizedresponseretrier_test.go

```diff
@@ -0,0 +1,52 @@
+package proxy
+
+import (
+	"github.com/kyma-project/kyma/components/proxy-service/internal/apperrors"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"testing"
+)
+
+func TestForbiddenResponseRetrier_CheckResponse(t *testing.T) {
```

Comment:
We are missing two cases here:
- should retry if status code is 401/403
- should not retry if it already did and still received 401/403